reporter/gcppubsub/pubsub_test.go

```diff
@@ -0,0 +1,99 @@
+package gcppubsub
+
+import (
+	"context"
+	"fmt"
+	"github.com/openzipkin/zipkin-go/model"
+	"os"
+	"sync"
+	"testing"
+	"time"
+
+	"cloud.google.com/go/pubsub"
+)
+
+var topicID string
+
+var once sync.Once // guards cleanup related operations in setup.
+
+func setup() *pubsub.Client {
+	ctx := context.Background()
+	proj := os.Getenv("GOOGLE_CLOUD_PROJECT")
+	fmt.Printf("GCP Project: %s\n", proj)
+	topicID = "test-topic"
+
+	client, err := pubsub.NewClient(ctx, proj)
+	if err != nil {
+		fmt.Printf("failed to create client: %s\n", topicID)
+		return nil
+	}
+
+	_, err = client.CreateTopic(ctx, topicID)
+	if err != nil {
```

Comment:
I think the best here is that instead of returning a `nil` we pass the `t *testing.T` variable into the setup and fatal (`t.Fatal`) if any of this `err`s is not `nil`. What do you think?

Reply:
It was like that originally. But the problem is that it would always fail in a CI server or everywhere where GOOGLE_CLOUD_PROJECT is not configured. As the test topic is created in an exiting GCP project... We we test the error but cannot test the success setup...